### PR TITLE
[IMP] web: resolve color field for file selection button

### PR DESCRIPTION
**Pull request by paru-odoo**
This PR aims to improve the color of the file input field that did not change when the theme color or input color was updated. With this PR, the file input field color will now adjust according to the background color of the input field and the theme color.

task-4174112

### CODE REVIEW

**Comment by msh-odoo** [comment]
@paru-odoo We can not fix it in Bootstrap lib files, even if we want to fix in bootstrap then we should create separate scss file in /static/src/libs/bootstrap/bootstrap_overridden.scss and do the fix there, Thanks.

File: addons/mrp_subcontracting/__manifest__.py [hunk 1]
```diff
@@ -93,7 +93,8 @@
             'web/static/lib/bootstrap/js/dist/scrollspy.js',
             'web/static/lib/bootstrap/js/dist/tab.js',
             'web/static/lib/bootstrap/js/dist/toast.js',
-            'web/static/src/libs/bootstrap.js',
+            'web/static/src/libs/bootstrap/bootstrap.js',
+            'web/static/src/libs/bootstrap/bootstrap.scss',
```
Comment (qsm-odoo) [inline addons/mrp_subcontracting/__manifest__.py:97]: I don't know what this change is about but surely it either deserves its own commit and comments or it's just wrong. There is a call to `_assets_bootstrap` a few lines below, surely this is already including bootstrap css?

Same for everywhere else you added this

File: addons/web/static/src/libs/bootstrap/bootstrap.scss [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.form-control {
+    // Overridden file input buttons theming
+    &::file-selector-button {
+        @include gradient-bg(darken($input-bg, 2%));
+    }
+
+    &:hover:not(:disabled):not([readonly])::file-selector-button {
+        background-color: darken($input-bg, 6%);
+    }
+}
```
Comment (qsm-odoo) [inline addons/web/static/src/libs/bootstrap/bootstrap.scss:10]: We already have a file for this kind of things, called `bootstrap_review.scss`

File: addons/web/static/src/libs/bootstrap/bootstrap.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.form-control {
+    // Overridden file input buttons theming
+    &::file-selector-button {
+        @include gradient-bg(darken($input-bg, 2%));
+    }
+
+    &:hover:not(:disabled):not([readonly])::file-selector-button {
+        background-color: darken($input-bg, 6%);
+    }
```
Comment (qsm-odoo) [inline addons/web/static/src/libs/bootstrap/bootstrap.scss:9]: Is this CSS based on something already existing in Bootstrap? I wonder the call the gradient-bg for the first rule but not for the second one?

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/web/static/src/libs/bootstrap/bootstrap.scss (hunk 2 above)
Comment (msh-odoo) [reply addons/web/static/src/libs/bootstrap/bootstrap.scss:10]: Thanks for drawing attention, will do the change

File: addons/web/static/src/libs/bootstrap/bootstrap.scss (hunk 3 above)
Comment (msh-odoo) [reply addons/web/static/src/libs/bootstrap/bootstrap.scss]: @qsm-odoo We have improved solution but we didn't get this comment, would you please elaborate?

Comment (paru-odoo) [reply addons/web/static/src/libs/bootstrap/bootstrap.scss]: @qsm-odoo We declared the variables $form-file-button-bg and $form-file-button-hover-bg in the bootstrap_overridden.scss and bootstrap_overridden_frontend.scss file. Since this variable files are loaded separately in backend and frontend assets, for that reason we made the changes in bootstrap_overidden and bootstrap_overidden_frontend.

**Comment by qsm-odoo** [comment]
@odoo/rd-design Can I have your opinion on this? :pray:

File: addons/web/static/src/libs/bootstrap/bootstrap.scss (hunk 3 above)
Comment (stefanorigano) [reply addons/web/static/src/libs/bootstrap/bootstrap.scss]: Hello 👋

`bootstrap_overridden.scss` is loaded and shared by **both** bundles, while `bootstrap_overridden_frontend.scss` and `bootstrap_overridden_backend.scss` are used **only** for the frontend and backend, respectively.  

**If** we wanted to achieve the same visual result in both bundles, modifying `bootstrap_overridden.scss` would have been enough... however we don’t want that because, in the backend, the button is normally managed by the `color-scheme: [light/dark]` rule 😉

 @paru-odoo ,  I suspect that the reason why you had to touch both files, it's because you forgot to add the [`!default` flags](https://sass-lang.com/documentation/variables/#default-values), thus the compilation was "failing" somewhere 😉
The flag  is also needed to allow pro-users to customise, so it should be there.

Thus, in conclusion: 
- define **frontend specific** overrides in `bootstrap_overridden_frontend.scss` only
- use the `!default` flag

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Thanks for your work @paru-odoo 👍
Here some remarks for you... the solution proposed was visually correct, nice touch. There is room for improvement on the assets management and consistency sides tho, even if in the second case we're talking about just minor discrepancies 😉

Thanks again!

**Comment by paru-odoo** [comment]
Hello @stefanorigano , changes have been made. We do the same thing that you suggested.
Thanks for your review :)

File: addons/web/static/src/scss/bootstrap_overridden_frontend.scss [hunk 4]
```diff
@@ -93,6 +93,10 @@ $form-range-thumb-bg: $body-bg !default;
 $form-range-thumb-active-bg: $primary !default;
 $form-range-thumb-disabled-bg: $input-disabled-bg !default;
 
+$btn-hover-bg-shade-amount: 15% !default;
```
Comment (stefanorigano) [inline addons/web/static/src/scss/bootstrap_overridden_frontend.scss:96]: Almost there 💪

By convention a `bootstrap_overridden*` file "_is a copy of the bootstrap variables.scss file where all the left-untouched variables definition have been removed_",  meaning that the **order** of the variable we define here should match BS default (see https://github.com/twbs/bootstrap/blob/main/scss/_variables.scss).

You should move `$btn-hover-bg-shade-amount`definition around line 68, in a new "section" called `// Buttons`, just before `// Forms`.

**Review by stefanorigano — CHANGES_REQUESTED** [review]

**Comment by paru-odoo** [comment]
Hey @stefanorigano , I made the changes.

**Review by stefanorigano — APPROVED** [review]
@paru-odoo , thanks, LGTM 👍 

@qsm-odoo ,  thanks a lot for the heads-up!

**Review by qsm-odoo — APPROVED** [review]
In fact, this was supposed to be a fix @paru-odoo, it was ok in 17.2 and broken depending on theme colors in 17.4. I retargeted.

@robodoo r+
